src/components/PopupWithForm.js

```diff
@@ -3,7 +3,7 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
     constructor(popupSelector, handleFormSubmit) {
         super({ popupSelector });
-        this._popupForm = document.querySelector('.modal__form');
+        this._popupForm = document.querySelector(`${popupSelector} .modal__form`);
```

Comment:
```js

        this._popupForm = document.querySelector(`${popupSelector} .modal__form`);

```

You don't need to search the popup and then the form here. It's a waste of resources


You need to search the form inside `this._popupElement` which is accessible from the parent class


```js

        this._popupForm = this._popupElement.querySelector(".modal__form");


```